tests/unit/amo/components/TestHeader.js

```diff
@@ -1,319 +1,47 @@
-    expect(link.prop('to')).toEqual('/collections/');
-  });
-
-  it('displays a view profile link when user is signed in', () => {
-    const id = 124;
-    const { store } = dispatchSignInActions({
-      userProps: { id },
-    });
-    const wrapper = renderHeader({ store });
-    const link = wrapper.find('.Header-user-menu-view-profile-link');
-
-    expect(link).toHaveLength(1);
-    expect(link.prop('children')).toEqual('View My Profile');
-    expect(link.prop('to')).toEqual(`/user/${id}/`);
-  });
-
-  it('displays an edit profile link when user is signed in', () => {
-    const { store } = dispatchSignInActions();
-    const wrapper = renderHeader({ store });
-    const link = wrapper.find('.Header-user-menu-edit-profile-link');
-
-    expect(link).toHaveLength(1);
-    expect(link.prop('children')).toEqual('Edit My Profile');
-    expect(link.prop('to')).toEqual('/users/edit');
-  });
-
-  it('allows a signed-in user to log out', () => {
-    const { store } = dispatchSignInActions();
-    const handleLogOut = sinon.stub();
-
-    const wrapper = renderHeader({ store, handleLogOut });
-
-    const onClick = wrapper.find('.Header-logout-button').prop('onClick');
-    onClick(createFakeEvent());
-
-    sinon.assert.calledWith(handleLogOut, {
-      api: store.getState().api,
-    });
-  });
-
-  it('displays the reviewer tools link when user has a reviewer permission', () => {
-    const { store } = dispatchSignInActions({
-      userProps: {
-        permissions: [ADDONS_REVIEW],
-      },
-    });
-    const wrapper = renderHeader({ store });
-    const link = wrapper.find('.Header-user-menu-reviewer-tools-link');
-
-    expect(link).toHaveLength(1);
-    expect(link.prop('children')).toEqual('Reviewer Tools');
-    expect(link).toHaveProp('href', '/reviewers/');
-  });
-
-  it('does not display the reviewer tools link when user does not have permission', () => {
-    const { store } = dispatchSignInActions();
-    const wrapper = renderHeader({ store });
-    const link = wrapper.find('.Header-user-menu-reviewer-tools-link');
-
-    expect(link).toHaveLength(0);
-  });
-
-  it('displays a "manage my submissions" link when user is logged in', () => {
-    const { store } = dispatchSignInActions();
-    const wrapper = renderHeader({ store });
-    const link = wrapper.find('.Header-user-menu-developers-submissions-link');
-
-    expect(link).toHaveLength(1);
-    expect(link.children()).toHaveText('Manage My Submissions');
-    expect(link).toHaveProp('href', '/developers/addons/');
-  });
-
-  it('displays blog, devhub and extension workshop links in the header on Firefox', () => {
-    const { store } = dispatchClientMetadata({
-      userAgent: userAgents.firefox[0],
+  const render = (props = {}) => {
+    return defaultRender(<Header {...props} />, {
+      store: dispatchClientMetadata({
+        clientApp: CLIENT_APP_FIREFOX,
+        userAgent: userAgents.chrome[0],
```

Comment:
No. We need this because we are testing that the `GetFirefoxBanner` component is not rendered when `forBlog` is `true`. If we did not have a `userAgent` that is not Firefox, then that banner would never render, and we'd have a false positive. This is one of many examples of where it gets quite confusing to write a test that relies on the behaviour of a component that is a child, but that's what we're stuck with without shallow rendering.